apps/prairielearn/src/pages/instructorQuestionSettings/instructorQuestionSettings.html.ts

```diff
@@ -0,0 +1,584 @@
+import { escapeHtml, html } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+import { z } from 'zod';
+
+import { Modal } from '../../components/Modal.html';
+import { IdSchema } from '../../lib/db-types';
+import { CourseWithPermissionsSchema } from '../../models/course';
+
+export const SelectedAssessmentsSchema = z.object({
+  title: z.string(),
+  course_instance_id: IdSchema,
+  assessments: z.array(
+    z.object({
+      assessment_id: IdSchema,
+      color: z.string(),
+      label: z.string(),
+    }),
+  ),
+});
+type SelectedAssessments = z.infer<typeof SelectedAssessmentsSchema>;
+
+type CourseWithPermissions = z.infer<typeof CourseWithPermissionsSchema>;
+
+interface SharingSet {
+  id: string;
+  name: string;
+  in_set: string;
+}
+
+export function InstructorQuestionSettings({
+  resLocals,
+  questionTestPath,
+  questionTestCsrfToken,
+  isEnterprise,
+  questionGHLink,
+  qids,
+  a_with_q_for_all_ci,
+  sharingEnabled,
+  sharingSetsIn,
+  sharingSetsOther,
+  editableCourses,
+  infoPath,
+}: {
+  resLocals: Record<string, any>;
+  questionTestPath: string;
+  questionTestCsrfToken: string;
+  isEnterprise: boolean;
+  questionGHLink: string | null;
+  qids: (string | null)[];
+  a_with_q_for_all_ci: SelectedAssessments[];
+  sharingEnabled: boolean;
+  sharingSetsIn: (SharingSet | null)[];
```

Comment:
When would we expect this array to contain `null` items? Did you mean this instead?

```suggestion
  sharingSetsIn: SharingSet[] | null;
```

That said, I expect we could simplify this just just always passing in an array, and just making sure it's empty if sharing isn't enabled.

Reply:
This array should not contain null items and should alway pass an array. Removed `null` in [e8cc26c](https://github.com/PrairieLearn/PrairieLearn/pull/9571/commits/e8cc26c79af51815e4e9bdbf5d82f9f432f5decc).